x-pack/legacy/plugins/lens/server/routes/field_stats.ts

```diff
@@ -0,0 +1,259 @@
+}
+
+export async function getNumberHistogram(
+  aggSearchWithBody: (body: unknown) => Promise<unknown>,
+  field: { name: string; type: string; esTypes?: string[] }
+): Promise<FieldStatsResponse> {
+  const searchBody = {
+    sample: {
+      sampler: { shard_size: SHARD_SIZE },
+      aggs: {
+        min_value: {
+          min: { field: field.name },
+        },
+        max_value: {
+          max: { field: field.name },
+        },
+        sample_count: { value_count: { field: field.name } },
+        top_values: {
+          terms: { field: field.name, size: 10 },
+        },
+      },
+    },
+  };
+
+  const minMaxResult = (await aggSearchWithBody(searchBody)) as AggregationSearchResponse<
+    unknown,
+    { body: { aggs: typeof searchBody } }
+  >;
+
+  const minValue = minMaxResult.aggregations!.sample.min_value.value;
+  const maxValue = minMaxResult.aggregations!.sample.max_value.value;
+  const terms = minMaxResult.aggregations!.sample.top_values;
+  const topValuesBuckets = bucketsToResponse<{ terms: { field: string; size: number } }>(terms);
+
+  let histogramInterval = (maxValue! - minValue!) / 10;
+
+  if (Number.isInteger(minValue!) && Number.isInteger(maxValue!)) {
+    histogramInterval = Math.ceil(histogramInterval);
+  }
+
+  if (histogramInterval === 0) {
+    return {
+      count: minMaxResult.hits.total,
+      sampleCount: minMaxResult.aggregations!.sample.sample_count.value!,
+      topValues: topValuesBuckets,
+      histogram: { buckets: [] },
+    };
+  }
+
+  const histogramBody = {
+    sample: {
+      sampler: { shard_size: SHARD_SIZE },
+      aggs: {
+        histo: {
+          histogram: {
+            field: field.name,
+            interval: histogramInterval,
+          },
+        },
+      },
+    },
+  };
+  const histogramResult = (await aggSearchWithBody(histogramBody)) as AggregationSearchResponse<
+    unknown,
+    { body: { aggs: typeof histogramBody } }
+  >;
+
+  return {
+    count: minMaxResult.hits.total,
+    sampleCount: minMaxResult.aggregations!.sample.sample_count.value!,
+    histogram: bucketsToResponse<typeof histogramBody['sample']['aggs']['histo']>(
+      histogramResult.aggregations!.sample.histo
+    ),
+    topValues: topValuesBuckets,
+  };
+}
+
+export async function getStringSamples(
+  aggSearchWithBody: (body: unknown) => unknown,
+  field: { name: string; type: string }
+): Promise<FieldStatsResponse> {
+  const topValuesBody = {
+    sample: {
+      sampler: { shard_size: SHARD_SIZE },
+      aggs: {
+        sample_count: { value_count: { field: field.name } },
+        top_values: {
+          terms: { field: field.name, size: 10 },
+        },
+      },
+    },
+  };
+  const topValuesResult = (await aggSearchWithBody(topValuesBody)) as AggregationSearchResponse<
+    unknown,
+    { body: { aggs: typeof topValuesBody } }
+  >;
+
+  return {
+    count: topValuesResult.hits.total,
+    sampleCount: topValuesResult.aggregations!.sample.sample_count.value!,
```

Comment:
The problem with this is that array fields are counted with the number of values, not just 1. This can lead to more than 100% in the UI because it's compared to the total number of hits:
<img width="630" alt="Screenshot 2019-09-13 at 11 04 25" src="https://user-images.githubusercontent.com/1508364/64852678-1b7a3b80-d61a-11e9-8d3e-02b899335e99.png">

Not sure about an easy way out of this, maybe we need another aggregation for that to explicitly check for empty fields?